### DISCUSSION TITLE
issues in migrating from wildfly keyclok17 to quarkus keycloak18

### DISCUSSION THREAD

**Discussion by kkcmadhu** [keycloak/keycloak#13680] [Archive]
i am trying to migrate from wildfly run time to quaruks and upgrading from wildfly keyclaok17 to quarkus keycloak 18.

When bring up keycloak18 against existing db, things work perfectly fine in my dev enviornment, however in stage keycloak 18 fails to start and i see that the update has not not happened (.MIGRATION_MODEL) still has keyclaok 17.0.1 as version.

I suspect the migration is taking some time and the transactions are being rolled back after 5 min.Not sure if there is a property to control this behaviour?

should i change the transaction timeout period to a higher value by teseting  quarkus.transaction-manager.default-transaction-timeout in quarkus.properties??

**Comment by kkcmadhu** [comment]
anwser my own question..
yes setting 
`quarkus.transaction-manager.default-transaction-timeout=PT20M`\
in
quarkus.properties helped me to complete the migration.
i had about 500+realms.

**Comment by vershnik** [comment]
Yes, transaction timeout is nice option to be able to configure in keycloak.conf.
